### PR TITLE
Feat/new range slider

### PR DESCRIPTION
**Pull request by iam4x**
⚠️ merge first https://github.com/algolia/instantsearch.js/pull/2107 I will re-open the PR after on the correct branch, this is just for keeping track of what is done and code review.

Use https://github.com/airbnb/rheostat as `range-slider` and should remove the usage of https://github.com/algolia/react-nouislider which is not meant to be maintained.

* fixes range slider issues with negatives values 🎉
* removes the "hacky" JS -> React bridge for nouislider
* tries to keep the same options as before for easy replacement for current client implementations
* fixes the wrong calculation of the "half" value (never got reported?)

### TODO:

* [x] POC
* [x] Tests
* [x] Remove legacy range-slider
* [x] Improve styles

### PREVIEW:

image

(price2)

### CODE REVIEW

File: src/components/Slider2/Pit.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import React, {PropTypes} from 'react';
+import cx from 'classnames';
+
+const Pit = ({style, children}) => {
```
Comment (Haroenv) [inline src/components/Slider2/Pit.js]: I don't really understand this name

File: src/widgets/index.js [hunk 2]
```diff
@@ -12,6 +12,7 @@ export {default as pagination} from '../widgets/pagination/pagination.js';
 export {default as priceRanges} from '../widgets/price-ranges/price-ranges.js';
 export {default as searchBox} from '../widgets/search-box/search-box.js';
 export {default as rangeSlider} from '../widgets/range-slider/range-slider.js';
+export {default as rangeSlider2} from '../widgets/range-slider2/range-slider2.js';
```
Comment (Haroenv) [inline src/widgets/index.js]: Why don't we deprecate the other slider and call that one rangeSliderOld or something, while giving this one the `rangeSlider` name?

File: src/components/Slider2/Pit.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+  return (
+    <div
+      style={ {...style, marginLeft: positionValue === 100 ? '-2px' : 0} }
+      className={ cx('ais-range-slider--marker ais-range-slider--marker-horizontal', {
+        'ais-range-slider--marker-large': shouldDisplayValue,
+      }) }
+    >
+      { shouldDisplayValue
+          ? <div className="ais-range-slider--value">{ Math.round(children * 100) / 100 }</div>
```
Comment (bobylito) [inline src/components/Slider2/Pit.js]: What about using [toFixed](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/toFixed) here ?

File: src/components/Slider2/Slider2.js [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+    ]),
+  }
+
+  handleChange = ({values}) => {
+    const {refine} = this.props;
+    refine(values);
+  }
+
+  // creates an array number where to display a pit point on the slider
+  computeDefaultPitPoints = memoize(({min, max}) => {
```
Comment (bobylito) [inline src/components/Slider2/Slider2.js]: I wouldn't go for memoization unless we hit an identified perf issue.

File: src/widgets/index.js (hunk 2 above)
Comment (iam4x) [inline src/widgets/index.js]: It's in the TODO 👍  It will be done right after the implementation is correct and tests are passing!

Comment (rayrutjes) [inline src/widgets/index.js]: Why don't we delete the other one all together given we will release a major version anyway?

File: src/components/Slider2/Pit.js (hunk 3 above)
Comment (iam4x) [inline src/components/Slider2/Pit.js]: from Slack conversation:

> .toFixed is not rounding to the superior decimal and is returning a string

image

File: src/components/Slider2/Slider2.js (hunk 4 above)
Comment (iam4x) [inline src/components/Slider2/Slider2.js]: Works for me, during my tests I was using the same min/max values so found it relevant

File: src/widgets/index.js (hunk 2 above)
Comment (iam4x) [inline src/widgets/index.js]: @rayrutjes Yes it's gonna be the case, like I said above and in the TODO :)

File: src/widgets/range-slider2/range-slider2.js [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+      templatesConfig: instantSearchInstance.templatesConfig,
+      templates,
+    });
+    return;
+  }
+
+  const shouldAutoHideContainer = autoHideContainer && min === max;
+
+  const minValue = start[0] === -Infinity ? min : start[0];
+  const maxValue = start[1] === Infinity ? max : start[1];
```
Comment (iam4x) [inline src/widgets/range-slider2/range-slider2.js]: Should we do this check directly into the React component?

cc @bobylito

Comment (bobylito) [inline src/widgets/range-slider2/range-slider2.js]: In the connector probably?

**Comment by iam4x** [comment]
Closing this PR for the new one opened on the correct branch 👍

**Pull request closed without merging**